### PR TITLE
Fixed missing and incorrect documentation.

### PR DESCRIPTION
**Pull request by Bilge**

### CODE REVIEW

File: lib/Doctrine/ODM/MongoDB/Mapping/ClassMetadataInfo.php [hunk 1]
```diff
@@ -1678,8 +1680,10 @@ public function setIdGeneratorType($generatorType)
 
     /**
      * Sets the Id generator options.
+     *
+     * @param array $generatorOptions Generator options.
      */
-    public function setIdGeneratorOptions($generatorOptions)
+    public function setIdGeneratorOptions(array $generatorOptions)
```
Comment (alcaeus) [inline lib/Doctrine/ODM/MongoDB/Mapping/ClassMetadataInfo.php]: As mentioned elsewhere, this is a BC break and needs to go.

**Comment by alcaeus** [comment]
@Bilge As mentioned in #1376 we can't change method signatures in minor releases, but I'd like to see the other improvements in the upcoming 1.1 release. Could you remove the method signature changes?

File: lib/Doctrine/ODM/MongoDB/SchemaManager.php [hunk 2]
```diff
@@ -445,8 +445,12 @@ public function createDocumentDatabase($documentName)
      * recreate the index, since dropDups only affects creation of
      * the unique index. Additionally, the background option is only
      * relevant to index creation and is not considered.
+     *
+     * @param array $mongoIndex Mongo index data.
+     * @param array $documentIndex Document index data.
+     * @return bool True if the indexes are equivalent, otherwise false.
      */
-    public function isMongoIndexEquivalentToDocumentIndex($mongoIndex, $documentIndex)
+    public function isMongoIndexEquivalentToDocumentIndex(array $mongoIndex, array $documentIndex)
```
Comment (Bilge) [inline lib/Doctrine/ODM/MongoDB/SchemaManager.php]: @alcaeus Is this one OK or does this have to go as well?

Comment (alcaeus) [inline lib/Doctrine/ODM/MongoDB/SchemaManager.php]: Signature change in a public method => out it goes ;)

**Comment by malarzm** [comment]
Thanks a bunch @Bilge for all latest changes! :+1:

**Comment by Bilge** [comment]
@malarzm @alcaeus It's not much, but this is the least I could do for the help you have both offered me recently on IRC, and I'm happy I could be of some help in return.

**Pull request merged**